### PR TITLE
fix(core): Use explicit types in configs to ensure valid decorator metadata

### PR DESCRIPTION
**Pull request by netroy**
## Summary
For the config decorators to work properly, all config fields need to have explicit types, otherwise `tsc` emits the `design:type` decorator metadata as `Object` instead of the actual type. This seems to be a bug in `tsc`.

This PR 
1. Adds a check to prevent us from creating any configs fields without explicit types, by throwing when config is being parsed, and
2. Updates all current config fields to use an explicit type.

## Review / Merge checklist

- [x] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [x] Tests included. 
- [x] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Review by ivov — APPROVED** [review]
LGTM. Noticed that numbers with quotes are parsed as numbers. Only minor comments, feel free to merge.

File: packages/@n8n/config/src/decorators.ts [hunk 1]
```diff
@@ -47,6 +47,11 @@ export const Config: ClassDecorator = (ConfigClass: Class) => {
 					} else {
 						value = value === 'true';
 					}
+				} else if (type === Object) {
+					// eslint-disable-next-line n8n-local-rules/no-plain-errors
+					throw new Error(
+						`Invalid decorator metadata on key "${key as string}" on ${ConfigClass.name}\n Please use explicit typing on all config fields`,
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/@n8n/config/src/decorators.ts:53]: I'm not sure if linking to the PR would help any more than this error does. considering that it's mostly you and I using this code, I think we can improve the UX here when more people start using this.

File: packages/@n8n/config/src/decorators.ts [hunk 2]
```diff
@@ -47,6 +47,11 @@ export const Config: ClassDecorator = (ConfigClass: Class) => {
 					} else {
 						value = value === 'true';
 					}
+				} else if (type === Object) {
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/@n8n/config/src/decorators.ts:50]: I can't see a scenario where the field type is `Object`, and not a decorated class instead.

**Comment by janober** [comment]
Got released with [`n8n@1.54.4`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.54.4)

**Pull request merged**